DOM_creation/Lesson3/main.js

```diff
@@ -0,0 +1,22 @@
+const ul = document.getElementById("js-ul");
+
+const a1 = document.createElement("a");
+a1.href = "a1.html";
+
+const a2 = document.createElement("a");
+a2.href = "a2.html";
+
+const img1 = document.createElement("img");
+img1.src = "/img/bookmark.png";
+
+const img2 = document.createElement("img");
+img2.src = "/img/message.png";
+
+const array = [{a: a1, img: img1, text: "a1"}, {a: a2, img: img2, text: "a2"}];
+
+array.forEach(e =>{
+  let li = document.createElement('li');
+  e.a.textContent= e.text;
+  li.appendChild(e.a).insertAdjacentElement('afterbegin', e.img);
+  ul.appendChild(li);
+});
```

Comment:
IMO
When adding multiple DOMs, you can use document.createDocumentFragment() for
It has better performance.

Document
https://developer.mozilla.org/en-US/docs/Web/API/Document/createDocumentFragment
```JavaScript
const fragment = document.createDocumentFragment();

array.forEach((e) => {
  let li = document.createElement("li");
  e.a.textContent = e.text;

  fragment.appendChild(li).appendChild(e.a).insertAdjacentElement("afterbegin", e.img);
});

ul.appendChild(fragment);
```

Reply:
Thank you so much for your suggestion! 
I did not know that there is a method like createDocumentFragment().
And thank you for sharing the link, it helped me to understand the concept better.

I appreciate that you also provided an example code